tests/integration/records/single_db/test_records_unload_datetime.py

```diff
@@ -0,0 +1,490 @@
+            }
+            uses_pandas = False
+            if self.engine.name == 'redshift':
+                if datetimeformattz != 'YYYY-MM-DD HH:MI:SSOF':
+                    # this is the only format supported by Redshift on
+                    # export
+                    uses_pandas = True
+            elif self.engine.name == 'mysql':
+                # mysql has no exporter defined, so everything is via Pandas
+                uses_pandas = True
+            elif self.engine.name == 'postgresql':
+                if datetimeformattz in ['YYYY-MM-DD HH:MI:SSOF',
+                                        'YYYY-MM-DD HH24:MI:SSOF']:
+                    # ensure we keep other areas of ISO style
+                    addl_hints = {
+                        'timeonlyformat': 'HH:MI:SS',
+                        'dateformat': 'YYYY-MM-DD',
+                        'datetimeformat': 'YYYY-MM-DD HH:MI:SS',
+                    }
+                else:
+                    uses_pandas = True
+            elif self.engine.name == 'vertica':
+                # Make sure our '\n' strings below are valid when comparing output
+                addl_hints['record-terminator'] = '\n'
+                if datetimeformattz not in ['YYYY-MM-DD HH:MI:SSOF',
+                                            'YYYY-MM-DD HH24:MI:SSOF']:
+                    uses_pandas = True
+            if uses_pandas:
+                # We're going to need to be sure to use hints that
+                # work in Pandas
+                addl_hints.update(pandas_compatible_addl_hints)
+                if 'AM' in datetimeformattz:
+                    # TODO: Add a GitHub issue for this
+                    logger.warning('Cannot export this dateformattz using Pandas or Redshift--'
+                                   'skipping test')
+                    continue
+
+            records_format = RecordsFormat(variant=VARIANT_FOR_DB[self.engine.name],
+                                           hints={
+                                               'datetimeformattz': datetimeformattz,
+                                               'compression': None,
+                                               'header-row': False,
+                                               **addl_hints,  # type: ignore
+                                           })
+            expect_pandas_failure = not self.has_pandas() and uses_pandas
+            try:
+                csv_text = self.unload(column_name='timestamptz',
+                                       records_format=records_format)
+                self.assertIn(csv_text, [create_sample(datetimeformattz) + "\n",
+                                         create_sample(datetimeformattz).replace('-00', '+00') + "\n",
+                                         create_sample(datetimeformattz).replace('-00', '') +
+                                         ".000000\n",
+                                         # TODO: Should this be necessary?
```

Comment:
Uncompleted punchlist item detected--consider resolving or moving this to your issue tracker